### PR TITLE
Allow component selectors without componentName

### PR DESCRIPTION
**Pull request by rgrjoh**

### CODE REVIEW

**Comment by simonsmith** [comment]
Be good if you could provide some context around this.

**Comment by rgrjoh** [comment]
See https://github.com/davidtheclark/stylelint-selector-bem-pattern/issues/15.

**Comment by davidtheclark** [comment]
@rgrjoh Thanks for the PR. Please add corresponding tests and this should be good to go.

**Comment by rgrjoh** [comment]
I’m afraid I don’t know where or how to add tests for this. Is there a suitable guide or tutorial somewhere?

**Comment by davidtheclark** [comment]
I think you'll want to add to https://github.com/postcss/postcss-bem-linter/blob/master/test/selector-validation.js, so have a look at what's going on in there and follow the patterns. In case you're unfamiliar, the `describe` and `it` functions are from the mocha test framework.

**Comment by rgrjoh** [comment]
Added tests. Pretty unfamiliar with unit tests, so I hope I got it right ;-).

File: test/selector-validation.js [hunk 1]
```diff
@@ -235,6 +235,47 @@ describe('selector validation', function() {
...
+      });
+      describe('as strings', function() {
+        runTests({
+          initial: '^\\.prefix$',
+          combined: '^\\.prefix-{componentName}(?:-[a-z]+)?$',
+        });
+      });
+
+      function runTests(componentSelectors) {
+        var patternB = { componentSelectors: componentSelectors };
```
Comment (davidtheclark) [inline test/selector-validation.js]: Is there a `patternA` somewhere?

**Comment by davidtheclark** [comment]
@rgrjoh Looks great! I just had one nitpicky comment, then I think we can merge this.

File: test/selector-validation.js (hunk 1 above)
Comment (rgrjoh) [inline test/selector-validation.js]: No, that variable name is just the result of copy & paste from one of the existing tests :-).

**Comment by rgrjoh** [comment]
Tests added. I thought I posted a message about it but it looks like I didn’t.

**Comment by davidtheclark** [comment]
@rgrjoh Thanks! I was on vacation. Will cut a new release now.

**Pull request merged**